walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -60,6 +65,28 @@ public boolean areAttendeesAvailable(Collection<String> attendees, Event event)
       return true;
   }
 
+    public Collection<TimeRange> updatedAvailableTimes(List<TimeRange> eventTimes, long requestDuration) {
+        TimeRange currentTimeRange = TimeRange.WHOLE_DAY;
+        Collection<TimeRange> availableTimes = new ArrayList<>();
+        Collections.sort(eventTimes, TimeRange.ORDER_BY_START);
+        for (TimeRange eventTime : eventTimes) {
+            if (currentTimeRange.overlaps(eventTime)) {
+                if (currentTimeRange.start() < eventTime.start() && eventTime.start() - currentTimeRange.start() >= requestDuration) // available time before event
+                {
+                    int duration = eventTime.start() - currentTimeRange.start();
+                    availableTimes.add(TimeRange.fromStartDuration(currentTimeRange.start(), duration));
+                }
+                if (eventTime.end() < currentTimeRange.end() && currentTimeRange.end() - eventTime.end() >= requestDuration) // available time after event
+                {
+                    int duration = currentTimeRange.end() - eventTime.end();
+                    currentTimeRange = TimeRange.fromStartDuration(eventTime.end(), duration);
+                }
+            }
```

Comment:
I think you might need to make sure there are no overlapping time ranges within eventTimes or in availableTimes at the end.

If `duration = 2`
`eventTimes = [(4,10),(6,10)] # after sort`
And if a day is from 0 to 10.
You might get `availableTimes = [(0,4),(0,6)]` Instead of just `availableTimes = [(0,6)]`